src/Microsoft.AspNetCore.Hosting/Startup/Startup.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Hosting
+{
+    public abstract class Startup : IStartup
```

Comment:
Having `Startup` as the base name is going to be an obnoxious conflict with app's `Startup`.